blocks/hero/hero.js

```diff
@@ -0,0 +1,126 @@
+import { decorateButtons, readBlockConfig } from '../../scripts/lib-franklin.js';
+
+/**
+ * Builds the image wrapper div from the block definition
+ *
+ * @param {HTMLDivElement} block the block
+ * @return {HTMLDivElement} the image wrapper
+ */
+function buildImageWrapper(block) {
+  const picture = block.querySelector('picture');
+  picture.querySelector('img').setAttribute('loading', 'eager');
+  const p = picture.parentElement;
+  const image = document.createElement('div');
+  image.classList.add('image');
+  image.append(picture);
+
+  if (p.nodeName.toLowerCase() === 'p' && !p.textContent.trim()) {
+    p.remove(); // remove floating/empty p
+  }
+  return image;
+}
+
+function buildProductContent(block) {
+  const config = readBlockConfig(block);
+  const keys = Object.keys(config);
+  const contentDiv = document.createElement('div');
+  contentDiv.classList.add('content');
+
+  ['logo', 'title', 'references', 'link'].forEach((part) => {
+    const idx = keys.indexOf(part);
+    if (idx >= 0) {
+      const content = block.children[idx].children[1];
+      let p = content.querySelector(':scope > p');
+      if (!p) {
+        p = document.createElement('p');
+      }
+      p.innerHTML = block.children[idx].children[1].innerHTML;
+      content.classList.add(part);
+      content.replaceChildren(p);
+      contentDiv.append(content);
+      const a = content.querySelector('a');
+      if (a) {
+        a.setAttribute('target', '_blank');
+      }
+    }
+  });
+  decorateButtons(contentDiv);
+  return contentDiv;
+}
+
+function buildCardContent(block) {
+  const config = readBlockConfig(block);
+  const keys = Object.keys(config);
+  const contentDiv = document.createElement('div');
+  contentDiv.classList.add('content');
+
+  if (config.title) {
+    const idx = keys.indexOf('title');
+    const content = block.children[idx].children[1];
+    content.classList.add('title');
+    contentDiv.append(content);
+  }
+
+  if (config.resources) {
```

Comment:
Is there an upper limit that you can think of where the number of elements would negatively impact performance when running this block?

Reply:
Do you mean number of variations? Not that i can think of, no.

Reply:
Cool, thanks!